Makefile

```diff
@@ -13,6 +13,7 @@ lint:
 	uv run --no-sync ruff check src tests examples || exit 1
 	uv run --no-sync --all-extras mypy src tests examples || exit 1
 	uv run --no-sync --all-extras pyright src tests examples || exit 1
+	uv run --no-sync --all-extras basedpyright examples -p examples/pyrightconfig.json || exit 1
```

Comment:
i think the problem is moreso that executing the examples from the venv of the main project simply doesnt work. it complains about relative paths and such. something like `uv run --directory examples/defer_import_slow_startup -- basedpyright -p ../pyrightconfig.json defer_import` fixes these issues